markdown/ch4.markdown

```diff
@@ -434,7 +434,7 @@ ax1.set_xlabel('Length of input');
 ```
 <!-- caption text="FFT execution time vs smoothness for different input lengths" -->
 
-The intuition is that, for smooth numbers, the FFT can be broken up
+The intuition is that, for smooth inputs, the FFT can be broken up
```

Comment:
The intent here is to describe the smoothness of the *number* of elements in the input, it's not an attribute of the *values* of the input.

Reply:
Funnily enough, I think we agree on the intended meaning - the reason that I felt changing the word "numbers" would be an improvement was because I thought it led the reader to consider individual values *within a sequence*, e.g. elements of an array, rather than a property of the input sequence itself.

Pursuant your comment it's clear that "inputs" doesn't get the job done either. After taking a closer look, I think the original wording is fine - any confusion that would arise from the text is probably mitigated by the examples.

Reply:
I think the point is that people should *not* consider the elements of the array, just the length of the array.  Whatever conveys that best!

Reply:
For smooth-sized inputs? ;)

Reply:
Or maybe it's best to reuse the terminology from when "smoothness" is introduced:

> (an input length is considered smooth when its largest prime factor is small)

So maybe "for smooth input lengths"

I think any of the suggestions works: all that is needed is an executive decision!

Reply:
Can't see anything wrong with that last one! I think it's the best of all the choices discussed until now.

Reply:
Smooth input lengths is 👍

Reply:
:+1: resolved in 8692583